clients/python/src/model_registry/models/artifact.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+from kiota_abstractions.serialization import ComposedTypeWrapper, Parsable, ParseNode, SerializationWriter
+from typing import Any, Callable, Dict, List, Optional, TYPE_CHECKING, Union
+
+if TYPE_CHECKING:
+    from .doc_artifact import DocArtifact
+    from .model_artifact import ModelArtifact
+
+@dataclass
+class Artifact(ComposedTypeWrapper, Parsable):
+    """
+    Composed type wrapper for classes DocArtifact, ModelArtifact
+    """
+    @staticmethod
+    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Artifact:
+        """
+        Creates a new instance of the appropriate class based on discriminator value
+        param parse_node: The parse node to use to read the discriminator value and create the object
+        Returns: Artifact
+        """
+        if not parse_node:
+            raise TypeError("parse_node cannot be null.")
+        try:
+            mapping_value = parse_node.get_child_node("artifactType").get_str_value()
+        except AttributeError:
+            mapping_value = None
+        result = Artifact()
+        if mapping_value and mapping_value.casefold() == "doc-artifact".casefold():
+            from .doc_artifact import DocArtifact
+
+            result.artifact_doc_artifact = DocArtifact()
```

Comment:
Here the properties are listed by many different names. Now, fwiw we can simply prefix the expected field names with `artifact_*`, but because the model doesn't define the required properties, we get errors anyway.